ui/components/multichain/create-named-snap-account/create-named-snap-account.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { KeyringTypes } from '@metamask/keyring-controller';
+import { InternalAccount } from '@metamask/keyring-api';
+import {
+  getNextAvailableAccountName as getNextAvailableAccountNameFromController,
+  setAccountLabel,
+} from '../../../store/actions';
+import { CreateAccount } from '..';
+import { Box, ModalHeader } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { getAccountName } from '../../../selectors';
+
+export type CreateNamedSnapAccountProps = {
+  /**
+   * Callback called once the account has been created
+   */
+  onActionComplete: (completed: boolean, reject?: boolean) => Promise<void>;
+
+  /**
+   * Address of the account to create
+   */
+  address: string;
+
+  /**
+   * Suggested account name from the snap
+   */
+  snapSuggestedAccountName?: string;
+};
+
+export const CreateNamedSnapAccount: React.FC<CreateNamedSnapAccountProps> = ({
+  onActionComplete,
+  address,
+  snapSuggestedAccountName,
+}) => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+
+  const onCreateAccount = async (name: string) => {
+    dispatch(setAccountLabel(address, name));
+    await onActionComplete(true);
+  };
+
+  const getNextAccountName = async (
+    accounts: InternalAccount[],
+  ): Promise<string> => {
+    // Get the name of the temporary internal account
+    let defaultAccountName: string = getAccountName(accounts, address);
+    console.log('accounts', JSON.stringify(accounts, null, 2));
+    if (!defaultAccountName) {
+      const nextAvailableName = await getNextAvailableAccountNameFromController(
+        KeyringTypes.snap,
+      );
+      // Format is "Snap Account <number>"
+      const parts = nextAvailableName.split(' ');
+      const accountNumber = parseInt(parts[parts.length - 1], 10);
+      if (isNaN(accountNumber)) {
+        defaultAccountName = nextAvailableName;
+      } else {
+        // Decrement the next available account number to get the temporary account name
+        parts[parts.length - 1] = (accountNumber - 1).toString();
```

Comment:
This logic seems a bit fragile, if the name format changes then this will break. If you only care about the "number" you can just get the length of the accounts array for the snap keyring

Reply:
Good idea! I thought that code was ugly and didn't realize we had a handy selector! I updated it here [18d6b06](https://github.com/MetaMask/metamask-extension/pull/25191/commits/18d6b069599701241d0f97702c39b75b9e2d8fb3)